src/main/java/com/tabnine/UninstallListener.kt

```diff
@@ -3,27 +3,20 @@ package com.tabnine
 import com.intellij.ide.plugins.IdeaPluginDescriptor
 import com.intellij.ide.plugins.PluginStateListener
 import com.intellij.openapi.diagnostic.Logger
-import com.intellij.util.text.SemVer
 import com.tabnine.binary.BinaryRequestFacade
 import com.tabnine.binary.requests.uninstall.UninstallRequest
 import com.tabnine.general.StaticConfig
-import com.tabnine.general.TabnineZipFile
-import com.tabnine.general.readTempTabninePluginZip
 import com.tabnine.lifecycle.UninstallReporter
-import java.nio.file.Paths
-import java.time.Duration
 
-private const val TABNINE_JAR_NAME = "TabNine-"
-private const val JAR_SUFFIX = ".jar"
-private val TABNINE_JAR_REGEX = """^$TABNINE_JAR_NAME\d+\.\d+\.\d+(-\w+\.\d+)?.jar""".toRegex()
+// An ugly workaround to distinguish between update and uninstall -->
+// If it's an update, this method name will appear in the stack trace.
+private const val updateMethodName = "installOrUpdatePlugin"
```

Comment:
maybe it's safer to check uninstallAndUpdateUi?

Reply:
why would it be better? im ok with either way

Reply:
because if the  function does not exists(moved/modified) you will not  trigger the uninstall flow

Reply:
oh nice one, will do